### DISCUSSION TITLE
Please improve the API for typing custom input's context object.

### DISCUSSION THREAD

**Discussion by KamilBeda** [formkit/formkit#1010] [General]
It's currently very hard to properly type `context` prop in custom vue input.
I think that it should be easier to "grab" all component specific props and builtin context features.
I tried to use augmented `FormKitInputProps` type to use when I define my custom input props with no success.

My augmentation:
```javascript
declare module '@formkit/inputs' {
  import { FormKitInputs, FormKitSlotData, FormKitBaseSlots } from '@formkit/inputs'

  interface FormKitInputProps<Props extends FormKitInputs<Props>> {
    vselect: {
      type: 'vselect'
      list: any[]
      appendTo?: string | HTMLElement
      detached?: boolean
      canClear?: boolean
      canDeselect?: boolean
      loading?: boolean
      object?: boolean
      createOption?: boolean
    }
  }
}

export {}
```

My props definition inside custom input:
```javascript
import type { FormKitFrameworkContext } from '@formkit/core'
import type { FormKitInputProps } from '@formkit/inputs'

const props = defineProps<{
  context: FormKitFrameworkContext & FormKitInputProps['vselect']
}>()
```
The above doesn't work because `FormKitInputProps` expects a generic. I tried to add required generic with the help from Justin Shroeder (in discord channel) like that `context: FormKitFrameworkContext & FormKitInputProps<{ type: 'vselect'; list: any[] }>` but this didn't help and my custom props were still infered as `unknown`.

Thank you for considering my proposal

**Comment by rikzwarthoff** [comment]
I'm having the same issue, can't figure out how to actually use the augmented Props in combination with defineProps in a custom component.
